testscript/testscript.go

```diff
@@ -434,6 +452,40 @@ Script:
 	}
 }
 
+func (ts *TestScript) applyScriptUpdates() {
+	if len(ts.scriptUpdates) == 0 {
+		return
+	}
+	for name, content := range ts.scriptUpdates {
+		found := false
+		for i := range ts.archive.Files {
+			f := &ts.archive.Files[i]
+			if f.Name != name {
+				continue
+			}
+			data := []byte(content)
+			if txtar.NeedsQuote(data) {
+				data1, err := txtar.Quote(data)
+				if err != nil {
+					ts.t.Fatal(fmt.Sprintf("cannot update script file %q: %v", f.Name, err))
+					continue
+				}
+				data = data1
+			}
+			f.Data = data
+			found = true
+		}
+		// Sanity check.
+		if !found {
+			panic("script update file not found")
+		}
+	}
+	if err := ioutil.WriteFile(ts.file, txtar.Format(ts.archive), 0666); err != nil {
```

Comment:
Why not 644?

Reply:
I think `0666` is the "right" value here? i.e. it's the value before `umask` is applied. Hence this function is akin to `os.Create`?

Reply:
Yeah, 666 is standard.